libsrc/Wi/bif_crypto.c

```diff
@@ -1635,15 +1649,19 @@ bif_get_certificate_info (caddr_t * qst, caddr_t * err_ret, state_slot_t ** args
       {
 	const unsigned char *s;
 	int i, n;
-	const ASN1_STRING *sig = cert->signature;
-	X509_ALGOR *sigalg = cert->sig_alg;
+	const ASN1_STRING *sig;
+	const X509_ALGOR *sigalg;
+	const ASN1_OBJECT *sig_alg_algorithm;
 	char buf[80];
 	caddr_t val;
 
-        i2t_ASN1_OBJECT(buf,sizeof (buf), sigalg->algorithm);
+	X509_get0_signature(&sig, &sigalg, cert);
+	X509_ALGOR_get0(&sig_alg_algorithm, NULL, NULL, sigalg);
+
+        i2t_ASN1_OBJECT(buf,sizeof (buf), sig_alg_algorithm);
 
-	n = sig->length;
-	s = sig->data;
+	n = ASN1_STRING_length(sig);
+	s = ASN1_STRING_get0_data(sig);
```

Comment:
Debian did not make these changes, but it doesn't really matter. Both should work.